### ISSUE TITLE
Scrolling up sometimes fails

### ISSUE THREAD

**Issue by dbrgn** [threema-ch/threema-web#18]
E.g. scrolling up does not work with the "home" key if conversation is scrolled all the way to the bottom.

I also heard reports that scrolling fails with a touchpad.

Most probably has to do with the angularjs-scroll-glue library.

**Comment by patrick-tee** [comment]
Confirmed: Touchpad scrolling up using a Samsung Galaxy Tab S2 jumps back to the end after half a second, no matter if the scrolling gesture is still in progress or if the finger is up again.

**Comment by lgrahl** [comment]
I'm also experiencing this on FF 55 (jumps back to the bottom if not scrolling further than a certain amount).

**Comment by mist** [comment]
This is really anoying, I'm using FF57 on Debian and current FF on Android and cannot see anything but the last few messages, because the site scrolls to the last message after less than a second.
Please fix this issue!

**Comment by rugk** [comment]
What do you mean with desktop? The chat overview where you can select the chat/contact?

**Comment by mist** [comment]
On my Desktop Computer. While on my Tablet Computer (Firefox@Android) I cannot use my cursor buttons because it hase none.

**Comment by studersi** [comment]
+1

I'm running Threema Web 1.8.2 on Linux Mint 18 in Firefox 58.0.2 (64-bit). It happens when I'm scrolling with my notebook's tablet as well as with my Logitech mouse.

**Comment by dbrgn** [comment]
@studersi does the problem also occur with Chromium? Maybe it is triggered by some smooth scrolling settings in Firefox...

**Comment by studersi** [comment]
@dbrgn You're right, it does seem to be related to smooth scrolling to an extent.

I don't have Chromium installed but I do have Google Chrome [Version 64.0.3282.186 (Official Build) (64-bit)]. It scrolls just fine on Chrome.

In Firefox, smooth scrolling is disabled. When I enable smooth scrolling, the problem disappears **almost** completely. It's still noticeable sometimes but it just seems to stutter before scrolling as planned. Very rarely, it stays at the bottom of the conversation.

**Comment by mist** [comment]
I can kind of confirm that It has to do with smooth scrolling.
Got a new machine with more power and without smooth scrolling set the problem kind of is there, although just annoying but not rendering the page useless. When I enable smooth scrolling the problem is gone.
On the Android tablet I see no way to enable smooth scrolling, so the problem persists.

**Comment by dbrgn** [comment]
If the last message in the conversation is an animgif that is playing, then scrolling up does not work in Firefox 58.

**Comment by heckenmann** [comment]
Changing line 39 in messenger.conversation.html to 
`<div id="conversation-chat" scroll-glue="!ctrl.showScrollJump" in-view-container ng-show="!ctrl.locked">`
seems to make the behavior a bit better.

**Comment by sebastianw** [comment]
On my Macbook scrolling with the touchpad also fails. Workaround: use the cursor key to scroll up a bit first, after that scrolling works with the touchpad as long as I don't scroll down to the end of the chat again. When I scroll back down to the end of the chat the problem occurs again.

**Comment by rugk** [comment]
Which browser is that?

**Comment by sebastianw** [comment]
Current Firefox on Mac

**Comment by dbrgn** [comment]
@heckenmann thanks, that improves things slightly :slightly_smiling_face: there's probably some issue with the underlying library that we need to debug.

**Comment by ventilator** [comment]
Scrolling up in a chat especially fails, if a) the phone is on LTE (no Wifi) and b) there is an image in the chat which is still loading. Loading takes about one minute, during that time, scrolling up is not possible (OS: Windows 10, Browser: Firefox). Kind of annoying, since I cannot read chat messages. 

Workaround: Setting Webbrowserscaling to 50%, so I can read messages further up while images are loading.

**Comment by dbrgn** [comment]
@ventilator note that you can always grab the scrollbar with the mouse to scroll.

**Comment by ventilator** [comment]
@dbrgn This won't help, grabbing and holding the scrollbar lets the interface jump up to the scroll bar position and down to the end of the thread with a frequency so fast, that reading is not possible

**Comment by ventilator** [comment]
1.8.2. But I tried, but cannot reproduce the effect at another PC at home, which has a faster internet connection. I will try to make a screencast of the effect once I have access again to the laptop at work which has a somewhat slower or at least filtered, firewalled corporate internet connection.

**Comment by mist** [comment]
@dbrgn I can also affirm that grabbing the scrollbar with the mouse did never work whenever I encountered this problem.

**Comment by rugk** [comment]
I could always grab the scrollbar with the mouse and move it "hard enough" to the top.
It's a bit like taming donkeys, but it works.

**Comment by rugk** [comment]
Some more insights (2.0.0-beta.7):
* when an image is still loading, you still have the chance to scroll up with your mouse wheel. However, you anyway have to try hard (i.e. it still tries to "stick to the ground")
* while loading, it resets the position to the bottom every 500ms or so – even if you've managed to scroll up
* when loading finished, the only way to scroll up is by using your mouse and grabbing the scroll bar as mentioned before

in short: it all seems like a demotivation campaign by annoying the user to stick to the recent stuff and not scroll back… :stuck_out_tongue_winking_eye:  

Fedora 28, Firefox 61

**Comment by ovalseven8** [comment]
Have this problem with a specific conversation right now (Threema Web 2.1.7; Firefox 66.0.1). Cannot scroll up. No images etc. to load in this conversation, just text. Actually, it's just a single message that does not fit on screen, so that the scroll bar is displayed.

**Comment by lgrahl** [comment]
> Actually, it's just a single message that does not fit on screen, so that the scroll bar is displayed.

That's a different issue then. Can you open a new one for it?

**Comment by dbrgn** [comment]
I don't think it's a different issue.

And I cannot reproduce it, neither in Firefox nor in Chrome.

**Comment by lgrahl** [comment]
> I don't think it's a different issue.

If there's a message filling the whole screen?

**Comment by ovalseven8** [comment]
Example:

Cannot scroll up:
![Screenshot_2019-03-26 Threema Web](https://user-images.githubusercontent.com/8258609/55011014-8f317900-4fe5-11e9-9e4c-6932c9db04ed.png)

Here's the single message that is above:
![Screenshot_2019-03-26 Threema Web(1)](https://user-images.githubusercontent.com/8258609/55011026-92c50000-4fe5-11e9-99f4-6130eaecf850.png)

**Comment by dbrgn** [comment]
Ah, from what you wrote...

> Actually, it's just a single message that does not fit on screen, so that the scroll bar is displayed.

...I assumed that the entire chat only contains a single multi-line message that is larger than what fits on the screen.

Thanks for the additional datapoint, but I see no indication that it's a different issue than what was discussed previously.
